src/proxy/momento/src/protocol/resp/zcount.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright 2025 Pelikan Foundation LLC.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use std::io::Write;
+use std::time::Duration;
+
+use momento::cache::{SortedSetLengthByScoreRequest, SortedSetLengthByScoreResponse};
+use momento::CacheClient;
+use protocol_resp::{SortedSetCount, ZCOUNT, ZCOUNT_EX, ZCOUNT_HIT, ZCOUNT_MISS};
+use tokio::time;
+
+use crate::error::ProxyResult;
+use crate::klog::{klog_1, Status};
+use crate::ProxyError;
+
+use super::{parse_score_boundary_as_float, update_method_metrics};
+
+pub async fn zcount(
+    client: &mut CacheClient,
+    cache_name: &str,
+    response_buf: &mut Vec<u8>,
+    req: &SortedSetCount,
+) -> ProxyResult {
+    update_method_metrics(&ZCOUNT, &ZCOUNT_EX, async move {
+        let (min_score, min_score_inclusive) = parse_score_boundary_as_float(req.min_score())?;
+        let (max_score, max_score_inclusive) = parse_score_boundary_as_float(req.max_score())?;
```

Comment:
This belongs in the request parsing since the scores have a special syntax. When that syntax is invalid, the request is invalid and should be reject at that level (here and elsewhere).

That way the code here can access `min_score` and `min_score_inclusive` from the parsed request object.